lib/internal/streams/operators.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const { AbortError } = require('internal/errors');
+const compose = require('internal/streams/compose');
+
+module.exports.map = function map(stream, fn) {
+  return compose(stream, async function* (source, { signal }) {
```

Comment:
Need some basic type checking on `stream` and `fn` with corresponding `ERR_INVALID_ARG` errors.

Reply:
The actual implementation is going to be refactored significantly, does the API look good?

Reply:
The API looks fine really. Are there any options that would make sense and justify adding an `options` argument following the `fn`?

Reply:
Well, `signal` (for the mapper) and `concurrency` for the `.map` function - we can call it highWaterMark or something similar but I suspect people won't understand what it means if we do ^^